src/workerd/io/worker-interface.capnp

```diff
@@ -157,6 +157,126 @@ struct Trace @0x8e8d911203762d34 {
   executionModel @25 :ExecutionModel;
   # the execution model of the worker being traced. Can be stateless for a regular worker,
   # durableObject for a DO worker or workflow for the upcoming Workflows feature.
+
+  # =====================================================================================
+  # Additional types for streaming tail workers
+
+  struct Attribute {
+    # An Attribute mark is used to add detail to a span over its lifetime.
+    # The Attribute struct can also be used to provide arbitrary additional
+    # properties for some other structs.
+    # Modeled after https://opentelemetry.io/docs/concepts/signals/traces/#attributes
+    struct Value {
+      inner :union {
+        text @0 :Text;
+        bool @1 :Bool;
+        int @2 :Int32;
```

Comment:
Getting the types right here is tricky – the tracing approach I had in mind was to support 64-bit integers but that doesn't fit into the double-based JS number type. If integers are limited to 32 bits we'll have to document it but it might just be the cleanest approach.
For float we should have Float64 though? That matches the JS number type.